docs/reference/l2.md

```diff
@@ -0,0 +1,48 @@
+# L2
+
+Within the Bristlemouth stack,
+the L2 layer is responsible for monitoring how the upper layers of the IP stack interacts with the 10BASE-T1L ethernet driver.
+The code itself interacts and follows the below diagram:
+
+```{image} l2.png
+:align: center
+:width: 1000
+:class: no-scaled-link
+```
+
+## Responsibilities
+
+The L2 layer is the glue between the network layer of the IP stack and the ethernet driver.
+It controls the flow of data to and from each of those layers.
+The following is an in depth explanation of how each of its responsibilities work:
+
+- Receive Events
+  - The ethernet driver will receive a packet and pass the raw data to `bm_l2_rx`
+  - Then creates an new buffer through the ip stack abstraction and copies the raw data from the packet to this buffer,
+  - Following,
+  it then queues up this data to be handled in the `bm_l2_thread`
+  - During handling,
+  if the packet is a multi-cast packet (a packet that is sent to a group of interested receivers),
+  it is transmitted down the wire to other nodes
+  - The packet is then submitted to the upper layers of the IP stack with `bm_l2_submit`,
+  so that it may processed
+- Transmit Events
+  - Transmission messages are sent through the IP stack from upper layers,
+  which is then fed through `bm_l2_link_output`
+  - `bm_l2_link_output` then preps the message,
+  utilizing the `bm_l2_prep` function,
+  and sends it to a queue which is handled in the `bm_l2_thread`
+  - When processed,
+  the message is then sent to the network driver to be transmitted on the line
+- Power Up/Down Events
+  - External modules can call the `bm_l2_netif_set_power` API to queue up a power event to turn on/off nodes downstream
+  - To note this will also kill communication the devices
```

Comment:
```suggestion
  - External modules can call `bm_l2_netif_set_power` to queue up a power event to turn on/off nodes downstream
  - Note: this will also kill communication to the downstream devices
```